gulpfile.js

```diff
@@ -4,109 +4,148 @@
         './tools/tasks/**/*.js',
-        PROJECT_PATH.js + '/**/*.js',
-        PROJECT_PATH.tests + '/**/*.js',
-        // exclude from linting
-        '!' + PROJECT_PATH.js + '/*.min.js',
-        '!' + PROJECT_PATH.js + '/**/*.min.js',
-        '!' + PROJECT_PATH.js + '/dist/*.js',
-        '!' + PROJECT_PATH.tests + '/coverage/**/*',
-        '!' + PROJECT_PATH.tests + '/unit/helpers/**/*',
-        '!' + PROJECT_PATH.tests + '/integration/*.bundle.js'
+        PROJECT_PATH.webpack + '*.config.js',
+        PROJECT_PATH.webpack + '/**/*.js',
+        '!' + PROJECT_PATH.webpack + '/*.min.js',
+        '!' + PROJECT_PATH.webpack + '/**/*.min.js',
     ],
     sass: [
-        PROJECT_PATH.sass + '/**/*.{scss,sass}'
-    ]
+        PROJECT_PATH.sass + '/**/*.{scss,sass}',
+        '!' + PROJECT_PATH.sass + '/libs/_svgsprite.scss',
+    ],
+    svg: {
+        icons: [PROJECT_PATH.svg + '/icons/**/*.svg'],
+        // other: [PROJECT_PATH.svg + '/other/**/*.svg'],
+    },
 };
 
-var DEFAULT_PORT = 8000;
-var PORT = parseInt(process.env.PORT, 10) || DEFAULT_PORT;
-var DEBUG = argv.debug;
-
-
 /**
- * Checks project deployment
+ * Generic utility to import gulp tasks and pass options to them
+ *
  * @param {String} id - task name
- * @returns {Object} - task which finished
+ * @param {Object} [extra] - optional options to pass
+ * @returns {Function} - task definition
  */
-function task (id) {
-    return require('./tools/tasks/' + id)(gulp, {
-        PROJECT_ROOT: PROJECT_ROOT,
-        PROJECT_PATH: PROJECT_PATH,
-        PROJECT_PATTERNS: PROJECT_PATTERNS,
-        DEBUG: DEBUG,
-        PORT: PORT,
-        argv: argv
-    });
+function task(id, extra) {
+    return require('./tools/tasks/' + id)(
+        gulp,
+        Object.assign(
+            {
+                PROJECT_ROOT: PROJECT_ROOT,
+                PROJECT_PATH: PROJECT_PATH,
+                PROJECT_PATTERNS: PROJECT_PATTERNS,
+                argv: argv,
+            },
+            extra
+        )
+    );
 }
 
-gulp.task('bower', task('bower'));
+
+// #############################################################################
+// TASKS
+
+/**
+ * WARNING: postcss-critical-split is considered experimental and may be buggy, so it's
+ * commented out by default
+ *
+ * Usage:
+ * - "gulp sass" (generates sass, splits the files, and injects the code)
+ * - "gulp sass --debug" (to generate unminified css with sourcemaps)
+ * - "gulp sass:compile" (just generates the base.css out of sass, handy to skip critical css)
+ * - "gulp sass:critical" (splits the base.css with the critical css)
+ * - "gulp sass:rest" (splits the base.css with the remaining "rest" css)
+ * - "gulp sass:inline" (injects the base-critical.css as inline css into the template)
+ */
+// gulp.task('sass', ['sass:critical', 'sass:rest', 'sass:inline']);
+gulp.task('sass', ['sass:compile']);
+gulp.task('sass:compile', task('sass/compile'));
+gulp.task('sass:critical', ['sass:compile'], task('sass/critical'));
```

Comment:
please comment this line and add above a comment that this is experimental and should be used with caution